File: problem4.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Mon Jan 21 21:24:26 2019

@author: meganhott
"""

# Starting program for Jan. 17

import numpy as np
import matplotlib.pyplot as plt

test=np.loadtxt("smallperiodictable.txt", dtype={'names': ('AN','name', 'symbol', 'AW','density','isotopes','year'), 'formats': ('i1', 'U16', 'U2','f4','f4','i4','i4')})

"Q1"
d = test['density']>1
d1 = np.count_nonzero(d == True)
density_l = test['name'][d]
print("There are", d1, "elements denser than water in their natural form. These elements are", density_l)
print()

"Q2"
i = test['isotopes']>100
i1 = np.count_nonzero(i == True)
isotopes_l = test['name'][i]
print("There are", i1, "elements with over 100 isotopes. These elements are", isotopes_l)
print()

"Q3"
j = test['year']<1900
j1 = np.count_nonzero(j == True)
year_l = test['name'][j]
print(j1,"elements were discovered before 1900. These elements are", year_l)
print()

"Q4"
plt.plot(test['AW'], test['density'])
plt.title("Density vs Atomic Weight")
plt.xlabel("Atomic Weight")
plt.ylabel("Density")

File: problem4.m
% Starting point for Jan. 17
load('smallperiodictable')

%Q1
%density(density>1)
%density_l = density > 1;
%density_l(density_l == 1)
%names(density_l == 1)
%dense_names = names(i) 

d = numel(density(density>1));
str1 = "There are %d elements denser than water in their natural form. These elements are";
sprintf(str1, d)

density_l = density > 1;
names(density_l == 1)

%Q2
i = numel(isotopes(isotopes>100));
str2 = "There are %d elements with over 100 isotopes. These elements are";
sprintf(str2, i)

isotopes_l = isotopes > 100;
names(isotopes_l)

%Q3
j = numel(discyear(discyear < 1900));
str3 = "%d elements were discovered before 1900. These elements are";
sprintf(str3, j)

discyear_l = discyear < 1900;
names(discyear_l)

%Q4
plot(atomicweight, density)
axis([0, 252, 0, 25])
title("Density vs Atomic Weight")
xlabel("Atomic Weight")
ylabel("Density")